pkg/operator/controller/ingress/status.go

```diff
@@ -775,10 +827,58 @@ func computeLoadBalancerStatus(ic *operatorv1.IngressController, service *corev1
 			Message: message,
 		})
 	}
-
 	return conditions
 }
 
+// computeLoadBalancerProgressingStatus returns the LoadBalancerProgressing
+// conditions for the given ingress controller which is used later to determine
+// the ingress controller's Progressing status.
+func computeLoadBalancerProgressingStatus(ic *operatorv1.IngressController, service *corev1.Service, platform *configv1.PlatformStatus) operatorv1.OperatorCondition {
+	// Compute the IngressControllerLoadBalancerProgressingConditionType condition for the LoadBalancer
+	if ic.Status.EndpointPublishingStrategy.Type == operatorv1.LoadBalancerServiceStrategyType {
+		switch {
+		case ic.Status.EndpointPublishingStrategy.LoadBalancer == nil:
+			return operatorv1.OperatorCondition{
+				Type:    IngressControllerLoadBalancerProgressingConditionType,
+				Status:  operatorv1.ConditionUnknown,
+				Reason:  "LoadBalancerStatusIncomplete",
+				Message: "status.endpointPublishingStrategy.loadBalancer is not set.",
+			}
+		case service == nil:
+			return operatorv1.OperatorCondition{
+				Type:    IngressControllerLoadBalancerProgressingConditionType,
+				Status:  operatorv1.ConditionTrue,
+				Reason:  "NoService",
+				Message: "LoadBalancer Service not created.",
+			}
+		}
+		wantScope := ic.Status.EndpointPublishingStrategy.LoadBalancer.Scope
+		haveScope := operatorv1.ExternalLoadBalancer
+		if IsServiceInternal(service) {
+			haveScope = operatorv1.InternalLoadBalancer
+		}
+		if wantScope != haveScope {
+			message := fmt.Sprintf("The IngressController scope was changed from %q to %q.", haveScope, wantScope)
+			switch platform.Type {
+			case configv1.AWSPlatformType, configv1.IBMCloudPlatformType:
+				message = fmt.Sprintf("%[1]s  To effectuate this change, you must delete the service: `oc -n %[2]s delete svc/%[3]s`; the service load-balancer will then be deprovisioned and a new one created.  This will most likely cause the new load-balancer to have a different host name and IP address from the old one's.  Alternatively, you can revert the change to the IngressController: `oc -n openshift-ingress-operator patch ingresscontrollers/%[4]s --type=merge --patch='{\"spec\":{\"endpointPublishingStrategy\":{\"loadBalancer\":{\"scope\":\"%[5]s\"}}}}'", message, service.Namespace, service.Name, ic.Name, haveScope)
+			}
+			return operatorv1.OperatorCondition{
+				Type:    IngressControllerLoadBalancerProgressingConditionType,
+				Status:  operatorv1.ConditionTrue,
+				Reason:  "ScopeChanged",
+				Message: message,
+			}
+		}
+	}
+	return operatorv1.OperatorCondition{
```

Comment:
Same comment as above, should we omit this condition if it is false, or better to be verbose?

Reply:
Better to set it False, since it is valuable information we don't want to lose track of.